sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -46,7 +47,9 @@ case class UnresolvedNamedLambdaVariable(nameParts: Seq[String])
     nameParts.map(n => if (n.contains(".")) s"`$n`" else n).mkString(".")
 
   override def exprId: ExprId = throw new UnresolvedException("exprId")
-  override def dataType: DataType = throw new UnresolvedException("dataType")
+  override def dataType: DataType = throw new SparkRuntimeException(
+    errorClass = "INVALID_LAMBDA_FUNCTION_CALL.NON_HIGHER_ORDER_FUNCTION",
```

Comment:
Could you clarify, please, why do you propose the error condition. IMHO, it is related to the condition when Spark expects a higher-order function but got something else.

Looking at the test, you are trying to fix another situation when a higher-order functions is not expected at all.